NumLib/ODESolver/EquationSystem.h

```diff
@@ -0,0 +1,42 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_EQUATIONSYSTEM_H
+#define NUMLIB_EQUATIONSYSTEM_H
+
+namespace NumLib
+{
+
+//! \addtogroup ODESolver
+//! @{
+
+//! Collection of basic methods every equation system must provide.
+class EquationSystem
+{
+public:
+    //! Return the number of equations.
+    virtual std::size_t getNumEquations() const = 0;
+
+    /*! Check whether this is actually a linear equation system.
+     *
+     * \remark
+     * Depending on its parameters an in general nonlinear equation system
```

Comment:
No, that's what i mean: The equation system is nonlinear "in general", but in some special case it can be linear, e.g., for constant coefficients.